src/metabase/warehouse_schema/models/table.clj

```diff
@@ -72,10 +108,17 @@
 
 (t2/define-before-update :model/Table
   [table]
-  (let [changes (t2/changes table)
+  (let [changes        (t2/changes table)
         original-table (t2/original table)
         current-active (:active original-table)
-        new-active (:active changes)]
+        new-active     (:active changes)]
+
+    ;; Prevent setting data_authority back to unconfigured once configured
+    (when (and (not= (keyword (:data_authority original-table :unconfigured)) :unconfigured)
+               (= (keyword (:data_authority changes)) :unconfigured))
+      (throw (ex-info "Cannot set data_authority back to unconfigured once it has been configured"
+                      {:status-code 400})))
```

Comment:
Possibly we should make an exception here for `:unknown`. If we got an unknown value, is it valid to move it back to `:unconfigured`? I'm not certain about the intended semantics here.

Reply:
I don't think we'll need this, but can always revisit. I think the idea is to force the user to pick the first time they edit a table's metadata.